src/realm/obj.cpp

```diff
@@ -39,6 +40,35 @@
 
 namespace realm {
 
+ConstObj ObjLink::get_obj(const Group& g) const
+{
+    return g.get_table(m_table_key)->get_object(m_obj_key);
+}
+
+Obj ObjLink::get_obj(Group& g) const
+{
+    auto target_table = g.get_table(m_table_key);
+    ObjKey key = get_obj_key();
+    ClusterTree* ct = key.is_unresolved() ? target_table->m_tombstones.get() : &target_table->m_clusters;
+    return ct->get(key);
+}
+
+void ObjLink::validate(const Group& g) const
```

Comment:
Same. Validation of an ObjLink should be something Group can do, not a method on ObjLink itself.